packages/blocks/src/api/raw-handling/phrasing-content-reducer.js

```diff
@@ -56,5 +56,26 @@ export default function phrasingContentReducer( node, doc ) {
 			node.removeAttribute( 'target' );
 			node.removeAttribute( 'rel' );
 		}
+
+		// Handle footnote and endnote links
+		// Save anchor elements' name attribute as id
+		if ( node.name && ! node.id ) {
+			node.id = node.name;
+		}
+		// Remove name attribute in any case: the use of the name attribute in a elements is obsolete: https://html.spec.whatwg.org/multipage/obsolete.html#obsolete-but-conforming-features
+		node.removeAttribute( 'name' );
+
+		// Search for common footnote and endnote id structure (Word, LibreOffice). If doesn't match, remove id attribute.
+		if (
+			node.id &&
+			! /^_?(?:ftn|edn|sdfootnote|sdendnote)/i.test( node.id )
```

Comment:
Not sure if I was clear before, but I don't agree with this part. Could we check `node.ownerDocument.querySelector('[href=${ node.id }]')` instead so we know if there's actually a link to this element or not? If no link exists to the element with this ID, then we should not keep the ID around.

Reply:
The philosophy of the paste filters is to keep them as general as possible and not start matching identifiers from specific platforms.

Reply:
Thank you for clarifying this and for the guidance, I totally get what you mean now. Will try to make it work.